bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/IContentAssistProcessorExtension2.java

```diff
@@ -0,0 +1,113 @@
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pauła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Extension interface of {@link IContentAssistProcessor} that forward information about
+ * autoActivation
+ *
+ * @since 3.24
+ */
+public interface IContentAssistProcessorExtension2 extends IContentAssistProcessor {
+
+	/**
+	 * Returns a list of completion proposals based on the specified location within the document
+	 * that corresponds to the current cursor position within the text viewer.
+	 *
+	 * @param request completionAssitRequest {@link ContentAssistRequest}
+	 * @return an array of completion proposals or <code>null</code> if no proposals are possible
+	 */
+	ICompletionProposal[] computeCompletionProposals(ContentAssistRequest request);
+
+	/**
+	 * Returns a list of completion proposals based on the specified location within the document
+	 * that corresponds to the current cursor position within the text viewer.
+	 *
+	 * @param request completionAssitRequest {@link ContentAssistRequest}
+	 * @return an array of completion proposals or <code>null</code> if no proposals are possible
+	 */
+	IContextInformation[] computeContextInformation(ContentAssistRequest request);
+
+	/**
+	 * Returns a list of completion proposals based on the specified location within the document
+	 * that corresponds to the current cursor position within the text viewer.
+	 *
+	 * @param viewer the viewer whose document is used to compute the proposals
+	 * @param offset an offset within the document for which completions should be computed
+	 * @return an array of completion proposals or <code>null</code> if no proposals are possible
+	 */
+	@Deprecated
+	default ICompletionProposal[] computeCompletionProposals(ITextViewer viewer, int offset) {
+		return computeCompletionProposals(new ContentAssistRequest(viewer, offset, false, false));
+	}
+
+	/**
+	 * Returns a list of completion proposals based on the specified location within the document
+	 * that corresponds to the current cursor position within the text viewer.
+	 *
+	 * @param viewer the viewer whose document is used to compute the proposals
+	 * @param offset an offset within the document for which completions should be computed
+	 * @return an array of completion proposals or <code>null</code> if no proposals are possible
+	 */
+	@Deprecated
+	default IContextInformation[] computeContextInformation(ITextViewer viewer, int offset) {
+		return computeContextInformation(new ContentAssistRequest(viewer, offset, false, false));
+	}
+
+	static IContentAssistProcessorExtension2 adapt(IContentAssistProcessor processor) {
```

Comment:
You should better use an `AdapterFactory` for this purpose

Reply:
I just used same approach as IContentAssitProcessorExtension :) but I'll do as you suggest

Reply:
For now I didn't introduced extra IAdapterFactory, it feels to me like to much overhead

Reply:
I made it a little bit different and I hope more flexible:

1. Two inner methods inside ContentAssistant that just verify instance of (no extra classes)
2. Extra IContentAssistantExtension5 with "buildAssistantRequest"

Reply:
It is better to separate the concerns instead of cluttering the API it also makes it much easier if later adjustments are needed because the interfaces/classes do not really change, so even if there is a small overhead it is worth the effort.